### PR TITLE
Fix the snapshot publishing script

### LINKED ISSUE #32729: [Bug]: beam_Release_Python_NightlySnapshot workflow broken
### What happened?

The workflow is not publishing any nightly snapshots after April 13, 2024: https://github.com/apache/beam/actions/workflows/beam_Release_Python_NightlySnapshot.yml

### Issue Priority

Priority: 1 (data loss / total loss of function)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by shunping**
Since setuptools 69.3.0,  the distribution name of a package has been normalized according to https://packaging.python.org/en/latest/specifications/binary-distribution-format/#escaping-and-unicode to support PEP 625 (https://github.com/pypa/setuptools/commit/b0135f5097f32a27b7a14e2c6296ba14bcb4e10b). 

In our case, the source and  binary distribution name would be changed from `apache-beam` to `apache_beam`.

This PR is to fix the script for publishing python snapshot so that the file can be published to gcs.

fixes #32729

related thread: #30955

### CODE REVIEW

**Comment by shunping** [comment]
r: @Abacn

File: sdks/python/scripts/run_snapshot_publish.sh [hunk 1]
```diff
@@ -30,8 +30,10 @@ DEP_SNAPSHOT_FILE_NAME="beam-py-requirements-$time.txt"
 # and located under Gradle build directory.
 cd $WORKSPACE/sdks/python/build
 
-# Rename the file to be apache-beam-{VERSION}-{datetime}.tar.gz
-for file in "apache-beam-$VERSION*.tar.gz"; do
+# Rename the file to be apache_beam-{VERSION}-{datetime}.tar.gz
+# Notice that the distribution name of beam has been normalized as "apache_beam"
+# since setuptools 69.3.0.
+for file in "apache_beam-$VERSION*.tar.gz"; do
```
Comment (Abacn) [inline sdks/python/scripts/run_snapshot_publish.sh:36]: shall we use `apache?beam-$VERSION*` so both are supported?

Comment (shunping) [reply sdks/python/scripts/run_snapshot_publish.sh:36]: Good point, I thought we pinned the setuptools, but we do not.

Comment (shunping) [reply sdks/python/scripts/run_snapshot_publish.sh]: Done.

**Review by Abacn — APPROVED** [review]

**Pull request merged**